### PR TITLE
add typings for stevelacy/bump-regex

### PR DESCRIPTION
**Pull request by silkentrance**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.

### CODE REVIEW

**Comment by stevelacy** [comment]
Kinda got my name wrong lol

**Comment by silkentrance** [comment]
@stevelacy sorry about that, but this is a good one :grin: fixing right now.

**Comment by stevelacy** [comment]
no worries 😆

**Comment by silkentrance** [comment]
@stevelacy do you think that the PR captures the essence of the API of bump-regex or is it missing something? my adhoc tests showed that it works quite well, but you never know...

**Comment by stevelacy** [comment]
Not sure on what the test should do, if it needs to verify output or just ensure the typings work. I haven't created a typing test before

**Comment by silkentrance** [comment]
@stevelacy the tests just make sure that the typings will work during a tsc compilation run.
what I meant was the typings in index.d.ts. does the signature of ``function BumpRegex(...)...`` meet your requirements, along with the definitions found in the BumpRegex namespace, namely Callback, Options and Result. I tried to match these as closely as possible to the expected input arguments and the to be expected return results from either bump-regex or the callback.

**Comment by silkentrance** [comment]
@stevelacy is the callback expected to return no result, i.e. is it of type void? if so, then we could simplify the Callback type declaration to just ``...() => void``.

**Comment by stevelacy** [comment]
the callback should always return either the error as the first argument, or `(null, result)`

**Comment by silkentrance** [comment]
@stevelacy the callback is called with either (err) or (null, result), but does it return anything? And according to common best practice, the callback should not return anything as that result will be lost/ignored. I will fix the declaration ASAP, my fault.

**Comment by minestarks** [comment]
@stevelacy since you own the library and you seem to be engaged here 😄 would you consider [bundling these typings](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) in your project to have more control over them and to reduce maintenance burden here?

File: types/bump-regex/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (minestarks) [inline types/bump-regex/tsconfig.json]: Please `strictNullChecks: true` for all new packages

File: types/bump-regex/tslint.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
```
Comment (minestarks) [inline types/bump-regex/tslint.json]: Please don't turn off linting rules, they are there for good reason!

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by silkentrance** [comment]
linting was enabled and arising issues have been fixed - need to run a quick integration test though

**Review by minestarks — APPROVED** [review]

**Comment by silkentrance** [comment]
@minestarks Thank you. I am currently working on getting dtslint to a state that will allow package authors to directly include these typings into their package.

**Pull request merged**